pkg/planner/core/find_best_task.go

```diff
@@ -1627,6 +1628,60 @@ func skylinePruning(ds *logicalop.DataSource, prop *property.PhysicalProperty) [
 	return candidates
 }
 
+// hasOnlyEqualPredicatesInDNF checks if all access conditions in DNF form are equal predicates
+func (c *candidatePath) hasOnlyEqualPredicatesInDNF() bool {
+	// Exit if this isn't a DNF condition or has no access conditions
+	if !c.path.IsDNFCond || len(c.path.AccessConds) == 0 {
+		return false
+	}
+	// If the minimum number of access conditions required for DNF is more than 1, then each OR condition
+	// must have at least 1 equal predicates to satisfy the DNF requirement. Return true.
+	if c.path.MinAccessCondsForDNFCond > 1 {
```

Comment:
this shortcut might not be expected since I tweaked your test and get the following

```
result = tk.MustQuery("explain format='brief' select * from t where (a > 1 and b > 0) or (a > 2 and b < 5)")
require.Contains(t, result.Rows()[0][0], "IndexLookUp")
```
```
2025/08/14 15:01:45 IndexLookUp 2666.67 root  
├─Selection(Build) 2666.67 cop[tikv]  or(and(gt(test.t.a, 1), gt(test.t.b, 0)), and(gt(test.t.a, 2), lt(test.t.b, 5)))
│ └─IndexRangeScan 3333.33 cop[tikv] table:t, index:idx_a_b(a, b) range:(1,+inf], keep order:false, stats:pseudo
└─TableRowIDScan(Probe) 2666.67 cop[tikv] table:t keep order:false, stats:pseudo
```